Jetpack: fetch capabilities for the given siteIDs

leandroalonso opened the pull request:
### Description

This PR adds the feature to retrieve the Jetpack capabilities (such as Scan or Backup) for a set of siteIDs.

### To test

* Code review
* CI must be green :)

leandroalonso committed:
Adds a Jetpack Capabilities service to return jetpack capabilities for a given site

leandroalonso committed:
Bump version

leandroalonso committed:
Fix some descriptions

leandroalonso committed:
Make JetpackCapabilitiesServiceRemote open

momo-ozawa commented on WordPressKit/JetpackCapabilitiesServiceRemote.swift:
```diff
@@ -0,0 +1,36 @@
+
+        wordPressComRestApi.POST(path,
+                                 parameters: ["sites": siteIds] as [String: AnyObject],
+                                 success: {
+                                     response, _  in
```
[nits] args should be on the previous line
```suggestion
                                 success: { response, _  in
```

momo-ozawa commented on WordPressKit/JetpackCapabilitiesServiceRemote.swift:
```diff
@@ -0,0 +1,36 @@
+                                        jetpackCapabilities[blogId] = (capabilities["capabilities"] as? [String] ?? []) as AnyObject
+                                    }
+                                    success(jetpackCapabilities)
+                                 }, failure: {
+                                     error, _ in
```
[nits] args should be on the previous line
```suggestion
                                 }, failure: { error, _ in
```

momo-ozawa reviewed (approved):
LGTM with a few tiny nitpicks! 👍

leandroalonso committed:
Move arg one live above

leandroalonso committed:
Refactor: use the single endpoint with dispatch groups instead of the batch

The batch endpoint was meant to be a server-to-server endpoint, so I'm removing in favor of the single one

momo-ozawa commented on WordPressKit/JetpackCapabilitiesServiceRemote.swift:
```diff
@@ -0,0 +1,38 @@
+/// A service that returns the Jetpack Capabilities for a set of blogs
+open class JetpackCapabilitiesServiceRemote: ServiceRemoteWordPressComREST {
+
+    /// Returns a Dictionary of capabilities for each given siteID
+    /// - Parameters:
+    ///   - siteIds: an array of Int representing siteIDs
+    ///   - success: a success block that accepts a dictionary as a parameter
```
Thanks for the refactor from batch -> single!! 🙏

leandroalonso committed:
Use background thread instead of the main to avoid hitting the UI

leandroalonso committed:
Use DispatchQueue to avoid blocking the thread

leandroalonso committed:
Remove debug output

leandroalonso committed:
Bump version

leandroalonso committed:
Merge branch 'develop' into task/jetpack_capabilities
